### PR TITLE
#163969677 Register Candidate Endpoint

### PR DESCRIPTION
**Pull request by odaridavid**
### What does this PR do?
- Adds Candidate Enpoint and registers blueprint for candidate
- Adds Tests for Candidate View and model
- Adds default error handler for 400 request
- Removes unused schema file

### Description of task to be completed?
Have register candidate functionality where they can only be registered once
### Relevant PT Stories
[#163969677](https://www.pivotaltracker.com/n/projects/2241897)
### How Can This Be Manually Tested
1. Clone the repository and checkout to ft-admin-create-candidate-163969677
2. Create .env file and add in environment to be used based on config file and a db uri
3. create a virtual environment and using the env file created above start the environment with flask and use an api test client to test endpoints

### Screenshots
1. Register Candidate
screenshot from 2019-02-20 00-14-43

2. If Candidate Exists
screenshot from 2019-02-20 00-14-49

### CODE REVIEW

**Comment by muathendirangu** [comment]
On your response messages try and make them abit more descriptive so a user can understand the issue easily assume the user does not understand computer jargon

**Comment by odaridavid** [comment]
> On your response messages try and make them abit more descriptive so a user can understand the issue easily assume the user does not understand computer jargon

@Muart-C Ok ill work on that

File: api/__init__.py [hunk 1]
```diff
@@ -30,6 +31,14 @@ def method_not_allowed(e):
     return make_response(jsonify(error_response), 405)
 
 
+def invalid_request(e):
+    error_response = {
+        "error": "400 ERROR CHECK YOUR REQUEST",
```
Comment (meshack-mbuvi) [inline api/__init__.py]: The error message is not very descriptive

File: api/v2/models/candidate_model.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                              INNER JOIN parties ON candidates.party=parties._id
+                              INNER JOIN users ON candidates.candidate=users._id;"""
+            cursor.execute(select_query)
+            self.db_conn.commit()
+            candidate_registered = cursor.fetchall()
+            if len(candidate_registered) == 0:
+                return 'Empty'
+            return candidate_registered
+        except psycopg2.IntegrityError:
+            # Deals with non existent data on primary tables or existent conflicting data
```
Comment (meshack-mbuvi) [inline api/v2/models/candidate_model.py]: The comment is not necessary here

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: api/v2/models/candidate_model.py (hunk 2 above)
Comment (odaridavid) [reply api/v2/models/candidate_model.py]: Ok let me work on them and resolve them. @meshack-mbuvi

**Pull request merged**